edb/lang/graphql/types.py

```diff
@@ -238,11 +244,65 @@ def _make_generic_input_type(self, base, ops):
             fields={op: GraphQLInputObjectField(base) for op in ops},
         )
 
+    def define_generic_order_types(self):
+        self._gql_ordertypes['directionEnum'] = GraphQLEnumType(
+            'directionEnum',
+            values=OrderedDict(
+                ASC=GraphQLEnumValue(),
+                DESC=GraphQLEnumValue()
+            )
+        )
+        self._gql_ordertypes['nullsOrderingEnum'] = GraphQLEnumType(
+            'nullsOrderingEnum',
+            values=OrderedDict(
+                SMALLEST=GraphQLEnumValue(),
+                BIGGEST=GraphQLEnumValue(),
+            )
+        )
+        self._gql_ordertypes['Ordering'] = GraphQLInputObjectType(
+            'Ordering',
+            fields=OrderedDict(
+                dir=GraphQLInputObjectField(
+                    GraphQLNonNull(self._gql_ordertypes['directionEnum']),
+                ),
+                nulls=GraphQLInputObjectField(
+                    self._gql_ordertypes['nullsOrderingEnum'],
+                    default_value='SMALLEST',
+                ),
+            )
+        )
+
+    @lru_cache(maxsize=None)
+    def get_order_fields(self, typename):
```

Comment:
Are you sure it's a good idea to create an unbounded LRU cache?

Reply:
Actually that's a copy-paste remnant of early incarnations where there was more re-use of these helper function results. It sort of made sense for `get_fields`, but doesn't make sense for `get_order_fields` or `get_filter_fields` (which are now called just once during init). I think I'll remove that altogether as optimizations are not yet the focus here.